### PR TITLE
Qute - introduce Eval section

### LINKED ISSUE #12792: Qute data eval mechanism
**Description**
It would be nice to have some kind of eval mechanism in Qute so that a data is processed as part of the template with evaluation:

```
bar = "evaluated";
foo = "also {bar} as a template"
```

template:
```
this is {eval foo}
```

output to:
```
this is also evaluated as a template
```

**Implementation ideas**

We could check that `foo` doesn't use `eval` to avoid any dangerous implementation, or we could see that as the developer problem no? I mean you don't need qute to create infinite loops :)

### PR DESCRIPTION
**Pull request by mkouba**
- to parse and evaluate templates dynamically
- resolves #12792

### CODE REVIEW

**Comment by ia3andy** [comment]
Wow, that's pretty cool! Thanks @mkouba

**Review by ia3andy — APPROVED** [review]
Very nice!

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -865,6 +865,24 @@ Template inheritance makes it possible to reuse template layouts.
 
 NOTE: Section blocks can also define an optional end tag - `{/title}`.
 
+==== Eval Section
+
+This section can be used to parse and evaluate a template dynamically.
+The behavior is very similar to the <<include_helper>> but:
+
+1. The template contents is passed directly, i.e. not obtained via an `io.quarkus.qute.TemplateLocator`,
```
Comment (gsmet) [inline docs/src/main/asciidoc/qute-reference.adoc:873]:
```suggestion
1. The template content is passed directly, i.e. not obtained via an `io.quarkus.qute.TemplateLocator`,
```

**Review by gsmet — CHANGES_REQUESTED** [review]

**Review by gsmet — APPROVED** [review]
I squashed the commits, looks good now.

**Pull request merged**